template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/knox/KnoxGatewayConfigProvider.java

```diff
@@ -90,7 +90,11 @@
                     config.add(config(SIGNING_KEY_ALIAS, SIGNING_IDENTITY));
                     if (kerberosConfig.isPresent()) {
                         String domain = kerberosConfig.get().getDomain();
-                        config.add(config(GATEWAY_WHITELIST, "^/.*$;^https?://(.+." + domain + "):[0-9]+/?.*$"));
+                        if (generalClusterConfigs.getAutoTlsEnabled()) {
```

Comment:
This looks good, although I think we could refactor the getRoleConfigs method and add a few more junit tests

Reply:
Refactored and added tests.